src/classes/app.py

```diff
@@ -273,6 +273,12 @@ def gui(self):
         # Connect our exit signals
         self.aboutToQuit.connect(self.cleanup)
 
+        # Process any queued events
+        QApplication.processEvents()
```

Comment:
The idea here is to fully process the singleShot timers (for applying geometry and position state) before showing the window, to avoid the window "jumping" right after we show it.